packages/react/src/components/toggle.tsx

```diff
@@ -0,0 +1,203 @@
+      checked: false,
+      disabled: false,
+    },
+  },
+);
+
+export interface ToggleProps
+  extends Omit<VariantProps<typeof toggleVariants>, "checked"> {
+  defaultChecked?: boolean;
+  checked?: boolean;
+  onCheckedChange?: (checked: boolean) => void;
+  className?: string;
+  children?: React.ReactNode;
+}
+
+export const Toggle: React.FC<ToggleProps> = ({
+  defaultChecked = false,
+  checked: controlledChecked,
+  onCheckedChange,
+  size = "medium",
+  disabled = false,
+  className = "",
+  children,
+}) => {
+  const isControlled = controlledChecked !== undefined;
+  const [internalChecked, setInternalChecked] = React.useState(
+    isControlled ? controlledChecked : defaultChecked,
+  );
+  const id = React.useId();
+
+  React.useEffect(() => {
+    if (isControlled) {
+      setInternalChecked(controlledChecked);
+    }
+  }, [isControlled, controlledChecked]);
+
+  const checked = isControlled ? controlledChecked : internalChecked;
+  const isDisabled = disabled === true;
+
+  const handleChange = () => {
+    if (!isDisabled) {
+      const newChecked = !checked;
+      if (!isControlled) {
+        setInternalChecked(newChecked);
+      }
+      onCheckedChange?.(newChecked);
+    }
+  };
+
+  return (
+    <div className={clx("flex items-center", className)}>
+      {React.Children.map(children, (child) =>
+        React.isValidElement(child)
+          ? React.cloneElement(child, {
+              checked,
+              disabled: isDisabled,
+              size,
+              onChange: handleChange,
+              id,
+            } as any)
+          : child,
+      )}
+    </div>
+  );
+};
+
+export interface ToggleLabelProps {
+  children: React.ReactNode;
+  className?: string;
+  size?: "medium" | "large";
+  id?: string;
+}
+
+export const ToggleLabel: React.FC<ToggleLabelProps> = ({
+  children,
+  className,
+  size = "medium",
+  id,
+}) => {
+  return (
+    <label
+      htmlFor={id}
+      className={clx(
+        "text-txt-black-700",
+        size === "medium" ? "text-sm" : "text-lg leading-[26px]",
+        className,
+      )}
+    >
+      {children}
+    </label>
+  );
+};
+
+export interface ToggleThumbProps
+  extends Omit<ToggleProps, "children" | "defaultChecked" | "onCheckedChange"> {
+  onChange?: () => void;
+  id?: string;
+}
+
+export const ToggleThumb: React.FC<ToggleThumbProps> = ({
+  checked,
+  disabled,
+  size = "medium",
+  onChange,
+  className,
+  id,
+}) => {
+  const isDisabled = disabled === true;
+
+  return (
+    <label
+      htmlFor={id}
+      className={clx(
+        toggleVariants({ size, checked, disabled: isDisabled }),
+        className,
+      )}
+    >
+      <input
+        id={id}
+        type="checkbox"
+        className="sr-only"
+        checked={checked}
+        onChange={onChange}
+        disabled={isDisabled}
+      />
+      <span
+        className={thumbVariants({ size, checked, disabled: isDisabled })}
+      />
+    </label>
+  );
+};
```

Comment:
Conflicting label(s) + thumb. Huddle

Reply:
Also, does not address the left + right label positioning

Reply:
okay boss, solved.